lightning/src/events/mod.rs

```diff
@@ -1460,8 +1489,12 @@ pub enum Event {
 	HTLCHandlingFailed {
 		/// The channel over which the HTLC was received.
 		prev_channel_id: ChannelId,
-		/// Destination of the HTLC that failed to be processed.
-		failed_next_destination: HTLCDestination,
+		/// The type of HTLC that was handled.
+		handling_type: HTLCHandlingType,
+		/// The reason that the HTLC failed.
+		///
+		/// This field will be `None` only for objects serialized prior to LDK 0.2.0.
+		handling_failure: Option<HTLCHandlingFailureReason>
```

Comment:
Consulted @jkczyz our resident naming expert -- 

He suggested renaming `HTLCHandlingType` to `HTLCHandlingFailureType`, and renaming the fields within the event to `failure_type` and `failure_reason`. This would also imply possibly renaming the `HTLCHandlingFailureType::ReceiveFailed` to just `::Receive`, similarly for forwards. Reasoning being that the type of HTLC handling is moreso the type of failure, conceptually (Jeff feel free to correct me here). 

Thoughts on this naming scheme? cc @joostjager

Reply:
Sounds like an improvement to me.

Reply:
🫡  gone ahead and also changed `ForwardFailed` -> `Forward` with the same reasoning as `ReceiveFailed` -> `Receive`